extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -994,6 +994,12 @@ private static void producePersistenceUnitDescriptorFromConfig(
             storageEngineCollector.add(persistenceUnitConfig.dialect.storageEngine.get());
         }
 
+        // Discriminator Column
+        persistenceUnitConfig.discriminator.ignoreExplicitForJoined.ifPresent(
```

Comment:
Avoid using lambdas here (for performance purposes)

Reply:
This is in build time code, we allow using lambda there. We enforce this rule only in runtime code.

Reply:
Done!